### PR TITLE
[Dynamo] Inline class method

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89098

Fixes: https://github.com/pytorch/pytorch/issues/93416

Class will be treated as UserDefinedObject which will treat forward as UserFunction. Then, it will just inline the function body into the top level module
cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: torch/_dynamo/variables/user_defined.py [hunk 1]
```diff
@@ -140,6 +140,11 @@ def __str__(self):
     def python_type(self):
         return self.value_type
 
+    def as_python_constant(self):
+        if self.value_type is not types.MemberDescriptorType:
+            return self.value
+        return super().as_python_constant()
```
Comment (voznesenskym) [inline torch/_dynamo/variables/user_defined.py:146]: nit: special condition + affirmative check + early return reads cleaner

```suggestion
        if self.value_type is types.MemberDescriptorType:
            return super().as_python_constant()
        return self.value
```

File: torch/_dynamo/variables/builder.py [hunk 2]
```diff
@@ -397,7 +397,7 @@ def index_source(key):
             return SkipFilesVariable(
                 value, guards=make_guards(GuardBuilder.FUNCTION_MATCH)
             )
-        elif istype(value, (type, ABCMeta)):
+        elif istype(value, (type, ABCMeta)) and not issubclass(value, torch.nn.Module):
```
Comment (voznesenskym) [inline torch/_dynamo/variables/builder.py:400]: does isinstance work here?

**Review by voznesenskym — APPROVED** [review]

File: torch/_dynamo/variables/builder.py (hunk 2 above)
Comment (tugsbayasgalan) [reply torch/_dynamo/variables/builder.py:400]: value in this case is a class name, not instance.

File: torch/_dynamo/variables/user_defined.py (hunk 1 above)
Comment (tugsbayasgalan) [reply torch/_dynamo/variables/user_defined.py:146]: sounds good. Will do

**Pull request closed without merging**